perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -1016,9 +1016,19 @@ public static void sendValidationEmail(User user, String url, String email, int
 		String linkLocation = "/gui/";
 		try {
 			URL urlObject = new URL(url);
+			String domain = urlObject.getProtocol()+"://"+urlObject.getHost();
```

Comment:
This solves the issue for now. But we still rely on `referer` header to be sent. In the end of this dev cycle we want GUI to send us domain and authorization filter to the API, so just keep that in mind. I'm not sure, if it should or shouldn't be part of this pull-request.

Reply:
So now it should be possible to either pass referer or verification link url, authentication can be passed as query parameter too, the old way of appending it to path is still possible for now, might want to delete later?